extensions/iceberg/src/main/java/io/deephaven/iceberg/util/SchemaProviderInternal.java

```diff
@@ -12,22 +12,24 @@
  */
 class SchemaProviderInternal {
 
-    interface SchemaProviderImpl {
+    interface SchemaProviderImpl extends SchemaProvider {
         /**
          * Returns the schema for the given table based on this {@link SchemaProvider}.
          */
         Schema getSchema(Table table);
     }
 
     // Implementations of SchemaProvider
-    static class CurrentSchemaProvider implements SchemaProvider, SchemaProviderImpl {
+    enum CurrentSchemaProvider implements SchemaProviderImpl {
+        INSTANCE;
```

Comment:
I think it's useful to give this a more meaningful name in case it gets printed out `INSTANCE` is less useful than `CURRENT_SCHEMA`.